src/compiler/checker.ts

```diff
@@ -7246,6 +7246,30 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             const propertyName = getPropertyNameNodeForSymbol(propertySymbol, context);
             context.enclosingDeclaration = saveEnclosingDeclaration;
             context.approximateLength += symbolName(propertySymbol).length + 1;
+
+            if (propertySymbol.flags & SymbolFlags.Accessor) {
+                const writeType = getWriteTypeOfSymbol(propertySymbol);
+                if (propertyType !== writeType) {
+                    const getterDeclaration = getDeclarationOfKind<GetAccessorDeclaration>(propertySymbol, SyntaxKind.GetAccessor)!;
+                    const getterSignature = getSignatureFromDeclaration(getterDeclaration);
+                    typeElements.push(
+                        setCommentRange(

```

Comment:
Those two are using direct calls to `setCommentRange` instead of the existing `preserveCommentsOn` helper. The only added benefit of `preserveCommentsOn` is that it handles `JSDocPropertyTag`s. However, those only matter when serializing typedefs, as far as I can tell. We are serializing an "inferred" type of the object literal here - its property declarations can't be sourced from `@property`, right?